addons/hr_holidays/migrations/12.0.1.5/post-migration.py

```diff
@@ -0,0 +1,162 @@
+from openupgradelib import openupgrade
+from psycopg2.extensions import AsIs
+
+
+def fill_hr_holidays_dates(cr):
+    openupgrade.logged_query(
+        cr, """
+        UPDATE hr_holidays
+        SET date_from = create_date
+        WHERE date_from IS NULL
+        """
+    )
+    openupgrade.logged_query(
+        cr, """
+        UPDATE hr_holidays
+        SET date_to = create_date
+        WHERE date_to IS NULL
+        """
+    )
+
+
+def fill_hr_leave_type_validation_type(cr):
+    openupgrade.logged_query(
+        cr, """
+        UPDATE hr_leave_type
+        SET validation_type = 'both'
+        WHERE %s = TRUE
+        """, (AsIs(openupgrade.get_legacy_name('double_validation')), ),
+    )
+
+
+def fill_hr_leave_type_allocation_type(cr):
+    openupgrade.logged_query(
+        cr, """
+        UPDATE hr_leave_type
+        SET allocation_type = 'no'
+        WHERE %s = TRUE
+        """, (AsIs(openupgrade.get_legacy_name('limit')), ),
+    )
+
+
+def fill_hr_leave(env):
+    # In pre-migration the hr_leave table still doesn't exist
+    env.cr.execute(
+        """
+        ALTER TABLE hr_leave
+        ADD COLUMN {} integer;
+        """.format(openupgrade.get_legacy_name('holidays_id'))
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        INSERT INTO hr_leave (category_id, date_from, date_to, department_id,
+            employee_id, first_approver_id, holiday_status_id, holiday_type,
+            manager_id, meeting_id, name, notes, number_of_days,
+            payslip_status, report_note, second_approver_id, state, user_id,
+            create_uid, create_date, write_uid, write_date, %s)
+        SELECT category_id, date_from, date_to, department_id, employee_id,
+            first_approver_id, holiday_status_id, holiday_type, manager_id,
+            meeting_id, name, notes, number_of_days, payslip_status,
+            report_note, second_approver_id, state, user_id, create_uid,
+            create_date, write_uid, write_date, id
+        FROM hr_holidays
+        WHERE %s = 'remove'
+        """, (
+            AsIs(openupgrade.get_legacy_name('holidays_id')),
+            AsIs(openupgrade.get_legacy_name('type')),
+        ),
+    )
+    # fix parent_id:
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE hr_leave hl1
+        SET parent_id = hl2.id
+        FROM hr_holidays hh1
+        INNER JOIN hr_holidays hh2 ON hh1.parent_id = hh2.id
+        INNER JOIN hr_leave hl2 ON hh2.id = hl2.%s
+        WHERE hl1.%s = hh1.id
+        """, (
+            AsIs(openupgrade.get_legacy_name('holidays_id')),
+            AsIs(openupgrade.get_legacy_name('holidays_id')),
+        ),
+    )
+    # update holiday_id of resource.calendar.leaves:
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE resource_calendar_leaves rcl
+        SET holiday_id = hl.id
+        FROM hr_leave hl
+        WHERE rcl.%s IS NOT NULL AND rcl.%s = hl.%s
```

Comment:
Don't need the `IS NOT NULL` part

Reply:
```yml
ERROR: argument of AND must be type boolean, not type integer
```

Reply:
This shouldn't happen, starting that there shouldn't be any `AND` clause.